packages/react-jss/src/withStyles.js

```diff
@@ -113,7 +113,17 @@ const withStyles = <Theme>(styles: Styles<Theme>, options?: HOCOptions<Theme> =
         super(props)
 
         this.state = WithStyles.createState(props)
-        WithStyles.manage(props, this.state)
+        const context = props.jssContext
+        const {sheet} = this.state
+        if (sheet && context.registry) {
+          context.registry.add(sheet)
+        }
+      }
+
+      componentDidMount() {
+        if (this.state.sheet) {
+          WithStyles.manage(this.props, this.state)
```

Comment:
This is wrong. This would only after the first render meaning a flash of unstyled HTML!

We should check in the constructor if we render on the server and then not manage the sheet

Reply:
Thanks, I overlooked that.